src/libs/actions/App.js

```diff
@@ -325,9 +325,11 @@ function createWorkspaceAndNavigateToIt(policyOwnerEmail = '', makeMeAdmin = fal
             }
 
             if (shouldNavigateToAdminChat) {
-                Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(adminsChatReportID));
```

Comment:
I replace it with ```Navigation.dismissModal(adminsChatReportID);```, it still redirects to the admins room, and removes the current RHN modal as well

Reply:
As I mentioned in the proposal https://github.com/Expensify/App/issues/28098#issuecomment-1736167972: 

>  we can make sure that the current RIGHT_MODAL_NAVIGATOR screen is REPLACE by report screen, rather than PUSH a new report screen to the stack by using Navigation.dismissModal function:

Reply:
please give me the video screen on your side

Reply:
> it still redirects to the admins room, and removes the current RHN modal as well

What is your problem? It does not redirect to admin room? or it does not remove the current RHN?

Reply:
Please give me your video screen (with my code change) to debug

Reply:
Oh. Currently, when the user has no workspace, they can create a workspace like you did in your video: in this case, there is no RHN modal is displayed, so the dismissModal function will call the below logic so that it does not navigate to admin report https://github.com/Expensify/App/blob/05aa0c317e12a230977188dd5afb18cf4015bcb1/src/libs/Navigation/Navigation.js#L182
So I think we should create a param called "isTheFirstWorkspace". So the logic will be updated to:
```
...
  if (shouldNavigateToAdminChat) {
                if (isFirstWorkspace) {
                    Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(adminsChatReportID));
                } else {
                    Navigation.dismissModal(adminsChatReportID);
                }
            }
...
```

Reply:
@fedirjh just updated the logic to handle the case: Create a workspace using FloatingActionButton. Here is the result:

[Screencast from 03-10-2023 00:14:47.webm](https://github.com/Expensify/App/assets/141406735/21562551-91f1-432b-8672-46e55266ceb6)

Reply:
@fedirjh In your above staging video, pressing the browser back button will not only navigate back to the workspace list but also navigate back to the previous report screen (observe that in your video, the admin report screen is replaced by concierge screen). So I think we have an issue with the back button browser. Do you have any suggestions to solve it?

Reply:
@fedirjh In your above staging video, pressing the browser back button not only navigates back to the workspaces list but also replace the admin report by the concierge report.

Reply:
so what should be expected?

Reply:
@fedirjh Additionally, when we call ```Navigation.navigate...``` multiple times at once, for example, if user is in the report screen (href = /r/123456789) then call 
```
Navigation.navigate(/settings);
Navigation.navigate(/settings/workspaces);
```
 the browser`s history is just ```[/r/{reportID}, /settings, /settings/workspaces]```

Reply:
what do you think about this one https://github.com/Expensify/App/pull/28573#discussion_r1346785177 ? Should we use setTimeout for each navigation.navigate()

Reply:
or sorry. My bad. I have updated this comment https://github.com/Expensify/App/pull/28573#discussion_r1346785177

Reply:
@fedirjh do you have any suggestions for this one? The issue is that window.history does not sync with the react-navigation stacks

Reply:
@fedirjh based on the issue here https://github.com/Expensify/App/pull/28573#discussion_r1346785177, I just updated the PR to fix the error mentioned here https://github.com/Expensify/App/pull/28573#discussion_r1344820717

Reply:
### Result

https://github.com/Expensify/App/pull/28573#discussion_r1344820717

[Screencast from 06-10-2023 22:40:21.webm](https://github.com/Expensify/App/assets/141406735/5eb3e33a-1c35-4601-a850-0eb633885d7b)

[Screencast from 06-10-2023 22:39:00.webm](https://github.com/Expensify/App/assets/141406735/bffbef13-ee6e-4ab5-aee7-6c696324553c)